### DISCUSSION TITLE
Should I switch to Wezterm instead of kitty?

### DISCUSSION THREAD

**Discussion by daUnknownCoder** [wezterm/wezterm#6159] [Q&A]
### What Operating System(s) are you running on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Hyprland

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

I am using `fish` shell with `kitty` right now as I changed the config a few years ago and it just works, so I haven't touched it in a while, now I installed `WezTerm` and it looks cool and its `.lua` so as I am using `Neovim`, `.lua` is efficient for me...

BUT... why does `WezTerm` use so much `RAM` 😭?
![image](https://github.com/user-attachments/assets/9b4feb2e-5369-4760-94a0-6f3e0ce7c674)

`memtake` is my custom fish function which uses `pgrep` to get RAM used

<details><summary>Fish function</summary>

```fish
function memtake
    set process_name $argv[1]
    set total_mem 0

    # Get process IDs using pgrep
    set pids (pgrep $process_name)

    # Loop through each PID and sum the memory usage
    for pid in $pids
        if test -f /proc/$pid/status
            # Extract the memory usage in kB from the status file
            set mem (grep "VmRSS" /proc/$pid/status | awk '{print $2}')
            set total_mem (math "$total_mem + $mem")
        end
    end

    # Convert memory usage to MB and display the total
    echo (math "$total_mem / 1024")" MB"
end
```
</details> 

If this gets sorted out, I am ready to spend weeks and months on breaking and rebuilding my wez config...

**Comment by c02y** [comment]
The memory issue is no problem for me.

I need to replace kitty with other terminal like wezterm because every time I wakeup from hibernation, switching to kitty session will freeze my whole system for over 10 seconds. Other terminals or programs are working fine.

Details: https://github.com/kovidgoyal/kitty/issues/8142

**Reply by daUnknownCoder** [reply]
can you do me a favour? tell me what is `hibernation`? bruh like when i have low battery, my laptop sends me a notification to plug in and at 3% or something, it goes to hibernate -> which feels like an unforced unmanned shutdown, like next time i plug in and press power button, i have a startup from scratch like wth is wrong, it doesnt suspend to memory, nor hard disk, nor battery 

yep yep ik what u r thinking, it doesnt auto hibernate, i installed some package a long time ago whose name i dont remember which sends notifications at 30% 10% 8% 3-5%

**Reply by c02y** [reply]
> can you do me a favour? tell me what is `hibernation`? bruh like when i have low battery, my laptop sends me a notification to plug in and at 3% or something, it goes to hibernate -> which feels like an unforced unmanned shutdown, like next time i plug in and press power button, i have a startup from scratch like wth is wrong, it doesnt suspend to memory, nor hard disk, nor battery
> 
> yep yep ik what u r thinking, it doesnt auto hibernate, i installed some package a long time ago whose name i dont remember which sends notifications at 30% 10% 8% 3-5%

Hibernation, save status into hard disk, I'm using Linux, it means saving status to swap file in disk drive, once replug your power, the status is not lost. 
Sleep, save status into memory, once replug power, your status is gone. 

In your case, Ibelieve that your system is shutdown because of low battery.
